playbooks/byo/openshift-cluster/upgrades/v3_1_minor/upgrade.yml

```diff
@@ -0,0 +1,45 @@
+---
+- include: ../../../../common/openshift-cluster/upgrades/v3_1_minor/pre.yml
+  vars_files:
+  - "{{lookup('file', '../../../../byo/openshift-cluster/cluster_hosts.yml')}}"
+  vars:
+    g_etcd_hosts: "{{ groups.etcd | default([]) }}"
+    g_master_hosts: "{{ groups.masters | default([]) }}"
+    g_nfs_hosts: "{{ groups.nfs | default([]) }}"
+    g_node_hosts: "{{ groups.nodes | default([]) }}"
+    g_lb_hosts: "{{ groups.lb | default([]) }}"
+    openshift_cluster_id: "{{ cluster_id | default('default') }}"
+    openshift_deployment_type: "{{ deployment_type }}"
+- include: ../../../../common/openshift-cluster/upgrades/v3_1_minor/upgrade.yml
+  vars_files:
+  - "{{lookup('file', '../../../../byo/openshift-cluster/cluster_hosts.yml')}}"
+  vars:
+    g_etcd_hosts: "{{ groups.etcd | default([]) }}"
+    g_master_hosts: "{{ groups.masters | default([]) }}"
+    g_nfs_hosts: "{{ groups.nfs | default([]) }}"
+    g_node_hosts: "{{ groups.nodes | default([]) }}"
+    g_lb_hosts: "{{ groups.lb | default([]) }}"
+    openshift_cluster_id: "{{ cluster_id | default('default') }}"
+    openshift_deployment_type: "{{ deployment_type }}"
+- include: ../../../openshift-master/restart.yml
```

Comment:
@abutcher, @detiber, I couldn't include the restart playbook from within a role so I brook the upgrade playbook in to parts.  However, now there's a lot of duplication both in this file and in each playbook where we have to 'Evaluate host groups and gather facts'.  How could I avoid all the duplication?